modules/rostests/apitests/user32/RedrawWindow.c

```diff
@@ -86,3 +91,1084 @@ START_TEST(RedrawWindow)
+
+    testRWcompare.resultColorPre1 = 0x00FF0000;
+    testRWcompare.resultColorPre2 = 0x0000FF00;
+    testRWcompare.resultColorPre1 = 0x00FF0000;
+    testRWcompare.resultColorPost2 = 0x0000FF00;
+    InitRect(&testRWcompare.resultUpdateRect, 0, 500, 800, 600 );
+    testRWcompare.resultNeedsUpdate = FALSE;
+    testRWcompare.resultWmEraseGnd = FALSE;
+    testRWcompare.resultWmNcPaint = FALSE;
+    testRWcompare.resultPaintIndex = 2;
+    ok(0 == TestRedrawWindow2(&testRW, &testRWcompare),"Test33 fail");
+
+    testRW.testName = L"Test34";
+    testRW.flags = RDW_VALIDATE | RDW_NOCHILDREN;
+    testRW.useRegion = TRUE;
+    InitRect(&testRW.regRect, 0, 500, 800, 600 );
+    testRW.useRect = FALSE;
+    InitRect(&testRW.rectRect, 0, 0, 200, 200 );
+    testRW.forcePaint = TRUE;
+    testRW.testChild = TRUE;
+
+    testRWcompare.resultColorPre1 = 0x00FF0000;
+    testRWcompare.resultColorPre2 = 0x0000FF00;
+    testRWcompare.resultColorPost1 = 0x00FF0000;
+    testRWcompare.resultColorPost2 = 0x0000FF00;
+    InitRect(&testRWcompare.resultUpdateRect, 0, 500, 800, 600 );
+    testRWcompare.resultNeedsUpdate = FALSE;
+    testRWcompare.resultWmEraseGnd = FALSE;
+    testRWcompare.resultWmNcPaint = FALSE;
+    testRWcompare.resultPaintIndex = 1;
+    ok(0 == TestRedrawWindow2(&testRW, &testRWcompare),"Test34 fail");
+
+    // rect defined area tests
+    testRW.testName = L"Test35";
+    testRW.flags = 0;
+    testRW.useRegion = FALSE;
+    InitRect(&testRW.regRect, 0, 500, 800, 600 );
+    testRW.useRect = TRUE;
+    InitRect(&testRW.rectRect, 0, 500, 800, 600 );
+    testRW.forcePaint = TRUE;
+    testRW.testChild = FALSE;
+
+    testRWcompare.resultColorPre1 = 0x000000FF;
+    testRWcompare.resultColorPre2 = 0x000000FF;
+    testRWcompare.resultColorPost1 = 0x000000FF;
+    testRWcompare.resultColorPost2 = 0x000000FF;
+    InitRect(&testRWcompare.resultUpdateRect, 0, 0, 200, 200 );
+    testRWcompare.resultNeedsUpdate = FALSE;
+    testRWcompare.resultWmEraseGnd = FALSE;
+    testRWcompare.resultWmNcPaint = FALSE;
+    testRWcompare.resultPaintIndex = 1;
+    ok(0 == TestRedrawWindow2(&testRW, &testRWcompare),"Test35 fail");
+
+    testRW.testName = L"Test36";
+    testRW.flags = RDW_INVALIDATE | RDW_ERASENOW;
+    testRW.useRegion = FALSE;
+    InitRect(&testRW.regRect, 0, 500, 800, 600 );
+    testRW.useRect = TRUE;
+    InitRect(&testRW.rectRect, 0, 500, 800, 600 );
+    testRW.forcePaint = TRUE;
+    testRW.testChild = FALSE;
+
+    testRWcompare.resultColorPre1 = 0x000000FF;
+    testRWcompare.resultColorPre2 = 0x000000FF;
+    testRWcompare.resultColorPost1 = 0x000000FF;
+    testRWcompare.resultColorPost2 = 0x0000FF00;
+    InitRect(&testRWcompare.resultUpdateRect, 0, 500, 800, 600 );
+    testRWcompare.resultNeedsUpdate = TRUE;
+    testRWcompare.resultWmEraseGnd = FALSE;
+    testRWcompare.resultWmNcPaint = FALSE;
+    testRWcompare.resultPaintIndex = 2;
+    ok(0 == TestRedrawWindow2(&testRW, &testRWcompare),"Test36 fail");
```

Comment:
After this change, we would not see the number of the failed test directly in the listing, but only the error line. For debugging I find the original text more handy. If you don't see a major problem.

Reply:
Not a major problem, but (code / maintenance) bloat nonetheless: line number is enough to locate failing check.

Reply:
I was still thinking about it. And if the programmer in question is looking for a bug, orientation by test number is sometimes easier than by line number. If you insist, I can remove it, but I personally find more information better than less. If I were debugging the problems associated with these tests I would have to add the debugging information again.

Reply:
@SergeGautherie If @turican0 is running tests in a loop (because the stuff to be tested is stored in an array), then using a test number is kind of mandatory to know which one may have failed.

Reply:
> using a test number

Sure, that is what I suggested too. Why not simply pass a (line / test / whatever) number to `TestRedrawWindow2()`?

Reply:
I see the use of this trace information more as additional information for manual debugging. If they are missing in the bot dumps, it doesn't matter so much, if someone is interested in details they can run a manual test.
About displaying errors directly instead of trace - yes I can redo it, but do we want to have about 396 tests instead of 36? Isn't that too much?